packages/block-library/src/search/style.scss

```diff
@@ -53,6 +53,10 @@ $button-spacing-y: math.div($grid-unit-15, 2); // 6px
 .wp-block-search.wp-block-search__button-only {
 	.wp-block-search__button {
 		margin-left: 0;
+		// Prevent unintended text wrapping.
+		flex-shrink: 0;
+		// Ensure minimum input field width in small viewports.
+		max-width: calc(100% - 100px);
```

Comment:
Can you elaborate on how this calc math works? What are those 100px meant to capture? Is it the width of the confirm-button?

Reply:
This 100px is the minimum width of the input field when it opens.

![with-max-input](https://github.com/WordPress/gutenberg/assets/54422211/b254caf8-7d12-4338-a70a-12a1db5fd0a7)

Without this style, the input field would be extremely narrow in mobile viewports and the button would overflow.

![without-max-width](https://github.com/WordPress/gutenberg/assets/54422211/c91dadda-c722-4f4e-bceb-c6f761eead66)

There is no clear reason for the number 100px, but I believe some fixed value is necessary.

Reply:
Sounds good to me.